modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java

```diff
@@ -207,7 +207,9 @@ public void addObjectRelationshipMappingTableValues(
 		else {
 			_objectEntryLocalService.updateObjectEntry(
 				userId, primaryKey2,
-				HashMapBuilder.<String, Serializable>put(
+				HashMapBuilder.<String, Serializable>putAll(
+					_objectEntryLocalService.getValues(primaryKey2)
+				).put(
```

Comment:
The localized values is deleted and re-inserted during [updateObjectEntry](https://github.com/liferay/liferay-portal/blob/1fae55c17ae5fe85030dee13c3d4f0c9310cef2f/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java#L1534-L1535)
Then, through re-insert process, the logic expect the values map contain some value of the localized fields, otherwise nothing is inserted and the value is completely lost.